applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py

```diff
@@ -212,15 +212,35 @@ def Initialize(self):
 
         # Initialize the distance correction process
         self._GetDistanceModificationProcess().ExecuteInitialize()
-
+        self._GetDistanceModificationProcess().ExecuteInitializeSolutionStep()
+        
+        #Here the initial water volume of the system is calculated without considering inlet and outlet flow rate
+        self.initial_system_volume=KratosCFD.FluidAuxiliaryUtilities.CalculateFluidNegativeVolume(self.GetComputingModelPart())
+    
         # Instantiate the level set convection process
         # Note that is is required to do this in here in order to validate the defaults and set the corresponding distance gradient flag
         # Note that the nodal gradient of the distance is required either for the eulerian BFECC limiter or by the algebraic element antidiffusivity
         self._GetLevelSetConvectionProcess()
 
         KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__, "Solver initialization finished.")
 
+        self.mass_source = False
+        if self.settings["formulation"].Has("mass_source"):
+            self.mass_source = self.settings["formulation"]["mass_source"].GetBool()
```

Comment:
Very minor, but I'd put this before the `Logger` message. Just to ease debugging it case it eventually crashes.

Reply:
Modification done! Thanks!